lib/tasks/index.rake

```diff
@@ -109,7 +109,11 @@ task :preindex => :environment do
 
       content = blob_content[entry.sha]
       expand!(content, tag_files, blob_content, categories)
-      asciidoc = Asciidoctor::Document.new(content, attributes: {'sectanchors' => ''})
+      content.gsub!(/link:technical\/(.*?)\[(.*?)\]/) do |line|
+          x = /link:technical\/(.*?)\[(.*?)\]/.match(line)
+          line = "link:#{x[1]}\[#{x[2]}\]"
+        end
```

Comment:
This match should be removing the trailing `.html`, too, right? The link works with the `.html`, but we typically avoid it.

I also think you can dispense with the extra `match` inside the block and just access the captures as `$1` and `$2` directly. Or better yet, you can do it without a code block entirely.

So both suggestions together:

```ruby
content.gsub!(/link:technical\/(.*?).html\[(.*?)\]/, 'link:\1[\2]')
```

Reply:
Also, we'd want this hunk (and the one to import the technical files) duplicated for the `local_index` task, I think.

Reply:
This part was mainly a copy/paste of the one for `linkgit:` transformation. The functional style was more in case some more complex processing would be needed.

For local_index, there's still the factorization of the code to be done. But, we can duplicate, of course.